test/state.cpp

```diff
@@ -1069,3 +1069,50 @@ TEST_CASE_METHOD(RelatedGroupTest, "Reinitialize the group")
     REQUIRE(state == new_states[0]);
   }
 }
+
+TEST_CASE_METHOD(StateTest, "Parent Hash with Empty Left Subtree")
+{
+  // Create a group with 4 members
+  states.emplace_back(group_id,
+                      suite,
+                      leaf_privs[0],
+                      identity_privs[0],
+                      key_packages[0].leaf_node,
+                      ExtensionList{});
+
+  const auto adds = std::vector{
+    states[0].add_proposal(key_packages[1]),
+    states[0].add_proposal(key_packages[2]),
+    states[0].add_proposal(key_packages[3]),
+  };
+
+  auto [_commit0, welcome0, new_state_0] =
+    states[0].commit(fresh_secret(), CommitOpts{ adds, true, false, {} }, {});
+  silence_unused(_commit0);
+  states[0] = new_state_0;
+
+  for (size_t i = 1; i < 4; i++) {
+    states.push_back({ init_privs[i],
+                       leaf_privs[i],
+                       identity_privs[i],
+                       key_packages[i],
+                       welcome0,
+                       std::nullopt,
+                       {} });
+  }
```

Comment:
Since you're only going to use `states[2]` below this line, you could just initialize it directly.
```suggestion
  const auto state_2 = State{
    init_privs[i],
    leaf_privs[i],
    identity_privs[i],
    key_packages[i],
    welcome0,
    std::nullopt,
    {},
  };
```

Similarly, you don't actually need the `states` vector above, you could just define `state_0`.